circuits/cpp/src/aztec3/circuits/rollup/test_utils/utils.cpp

```diff
@@ -258,10 +258,8 @@ std::array<PreviousRollupData<NT>, 2> get_previous_rollup_data(DummyComposer& co
                                              kernel_data[i].public_inputs.end.new_commitments[j]);
         }
         auto contract_data = kernel_data[i].public_inputs.end.new_contracts[0];
-        auto contract_leaf = crypto::pedersen_commitment::compress_native(
-            { contract_data.contract_address, contract_data.portal_contract_address, contract_data.function_tree_root },
-            GeneratorIndex::CONTRACT_LEAF);
-        if (contract_data.contract_address != 0) {
+        auto contract_leaf = contract_data.hash();
```

Comment:
Do we need the `contract_leaf` for anything but the `update_element()`? Seems like it could be moved into the if or moved all the way in. 

Similar for some of the above.